src/main/java/com/idrsolutions/microservice/Individual.java

```diff
@@ -70,18 +72,32 @@ public void doError(final int errorCode, final String errorMessage) {
      * @return a JSON string representing this individuals state
      */
     String toJsonString() {
-        final StringBuilder json = new StringBuilder();
-        json.append("{\"state\":\"").append(state).append("\"")
-                .append(errorCode != null ? ",\"errorCode\":" + errorCode : "")
-                .append(errorMessage != null ? ",\"error\":\"" + errorMessage + "\"" : "");
-
-        for (final Map.Entry<String, JsonValue> valuePair : customValues.entrySet()) {
-            json.append(",\"").append(valuePair.getKey()).append("\":").append(valuePair.getValue().toString());
+        final JsonObjectBuilder json = Json.createObjectBuilder()
+                .add("state", state);
+        if (errorCode != null) {
+            json.add("errorCode", errorCode);
         }
+        if (errorMessage != null) {
```

Comment:
```suggestion
```

Reply:
Not against the core idea, I would make a minor adjustment. Thoughts on the final result of:
`if (errorCode != null || errorMessage != null) {`
`    json.add("errorCode", errorCode);`
`    json.add("error", errorMessage);`
`}`

Reply:
In theory if there is an error code there should always be an error message so you should only need to check for the error code.

Reply:
Good point, was trying to be safe where not needed.

Reply:
Have committed the change

Reply:
If I'm honest this was my mistake - I misread the two if statements as null-checking the same value.